public/js/app.js

```diff
@@ -0,0 +1,230 @@
+  fullPostContent.style.display = 'block';
+}
+
+// add click event to nav bar
+
+surfNav.addEventListener('click', sendRequest('surfing'));
+spaceNav.addEventListener('click', sendRequest('astrophotography'));
+earthNav.addEventListener('click', sendRequest('EarthPorn'));
+randomNav.addEventListener('click', randomSubRedditFunc);
+
+function randomSubRedditFunc() {
+  let randomSubArray = ['pics', 'hawaii', 'FoodPorn', 'askReddit', 'dogswithjobs', 'nba', 'javascript'];
+  let randomize = randomSubArray[Math.floor(Math.random() * randomSubArray.length)];
+
+  sendRequest(randomize)();
+}
+
+// // endless scroll attempt
+window.onscroll = function(ev) {
+  if (window.innerHeight + window.scrollY >= document.body.offsetHeight) {
+    sendRequestInfinite();
+  }
+};
+
+// search function below
+let searchBarBig = document.getElementById('searchBarBig');
+searchButtonBig.addEventListener('click', searchFunctionBig);
+searchButtonSmall.addEventListener('click', searchFunctionSmall);
+
+function searchFunctionBig() {
+  let input = searchBarBig.value;
+  sendRequest(input)();
+}
+
+function searchFunctionSmall() {
+  let input = searchBarSmall.value;
+  sendRequest(input)();
+}
+
+// user settings open
+
+let squareBox = document.getElementById('squareBox');
+squareBox.addEventListener('click', showUser);
+
+function showUser() {
+  postsBox.innerHTML = '';
+  userContainer.style.display = 'block';
+}
+
+let closeButton = document.getElementById('closeButton');
+closeButton.addEventListener('click', showPosts);
+
+function showPosts() {
+  userContainer.style.display = 'none';
+}
+
+let squareBoxSmall = document.getElementById('squareBoxSmall');
+squareBoxSmall.addEventListener('click', showUser);
+
+// slider settings function below
+
+settingsButton.addEventListener('click', showSettings);
+
+function showSettings() {
+  userContainer.style.display = 'none';
+  slideBox.style.display = 'block';
+}
+
+let slider = document.getElementById('fontSlider');
+var output = document.getElementById('testFont');
+output.innerHTML = slider.value + ' pixels';
+
+slider.oninput = function() {
+  output.innerHTML = this.value + ' pixels';
+  output.style.fontSize = `${this.value}px`;
+};
+
+changeFontButton.addEventListener('click', changeFontSize);
+
+function changeFontSize() {
+  let findContent = document.getElementsByClassName('navBar');
+  for (let i = 0; i < findContent.length; i++) {
+    findContent[i].style.fontSize = `${slider.value}px`;
+  }
+}
+
+backToSettings.addEventListener('click', backSetttings);
+
+function backSetttings() {
+  userContainer.style.display = 'block';
+  slideBox.style.display = 'none';
+}
+
+// background function
+
+changeBackground.addEventListener('click', switchBack);
+
+function switchBack() {
+  document.body.style.backgroundImage =
+    'url(https://previews.123rf.com/images/atomicchamp/atomicchamp1512/atomicchamp151200020/50324444-sun-and-cloud-background-with-a-pastel-colored-.jpg)';
+}
+
+sendRequest('surfing')();
```

Comment:
This works and I'll approve this, but you should look into [window.onload](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onload) instead of just firing the function at the end of your file. Window.onload will ensure that the function is fired when the html is ready.